rocketpy/mathutils/function.py

```diff
@@ -2612,6 +2622,77 @@ def compose(self, func, extrapolate=False):
                 extrapolation=self.__extrapolation__,
             )
 
+    def _check_user_input(
+        self,
+        source,
+        inputs,
+        outputs,
+        interpolation,
+        extrapolation,
+    ):
+        # check source for data type
+
+        # if list or ndarray, check for dimensions, interpolation and extrapolation
+        if isinstance(source, (list, np.ndarray)):
+            # this will also trigger an error if the source is not a list of numbers or if the array is not homogeneous
+            source = np.array(source, dtype=np.float64)
+
+            # check dimensions
+            source_dim = source.shape[1]
+
+            # check output type and dimensions
+            if isinstance(outputs, str):
+                outputs = [outputs]
+
+            elif len(outputs) > 1:
+                raise ValueError(
+                    f"Output must either be a string or have dimension 1, it currently has dimension ({len(outputs)})."
+                )
+
+            # check interpolation and extrapolation
+            if source_dim > 2:
+                # check for inputs and outputs
+                if inputs == ["Scalar"]:
+                    inputs = [f"Input {i+1}" for i in range(source_dim - 1)]
+                    warnings.warn(
+                        f"Inputs not set, defaulting to {inputs} for multidimensional functions.",
+                    )
+
+                if interpolation != None and interpolation != "shepard":
```

Comment:
```suggestion
                if interpolation not in [None , "shepard"]:
```

This is more direct and readable

Reply:
Updated

Reply:
I applied the requested changes but I don't have the information about the testes you said it didn't pass and the unit tests

Reply:
RocketPy adopts pytest suit for maintaining unit tests that cover small parts of the code. The GitHub actions are used to run the pytest module on every Pull Request. It's possible to check that the tests didn't pass: 

https://github.com/RocketPy-Team/RocketPy/pull/451/checks
![image](https://github.com/RocketPy-Team/RocketPy/assets/63590233/7786cc8d-f27d-40d4-aaa5-505ae4c2970d)

I'm going to fix it as soon as I understand where the bug is.